src/BootKeyboard/BootKeyboard.h

```diff
@@ -60,7 +60,36 @@ class BootKeyboard_ : public PluggableUSBModule {
   uint8_t getProtocol(void);
   void setProtocol(uint8_t protocol);
 
-  uint8_t default_protocol = HID_REPORT_PROTOCOL;
+  /* The USB specification says that keyboards should start with
+   * HID_REPORT_PROTOCOL as a default, and yet, we default to HID_BOOT_PROTOCOL
+   * instead. We do this to support operating systems that require the keyboard
+   * to default to boot protocol during negotiation in order to enable them to
+   * wake the host up. This includes Linux, MacOS, and Windows. If the keyboard
+   * conforms to the spec, and starts in report mode, it will work under these
+   * operating systems, but if they are suspended, the keyboard will not be
+   * allowed to wake them up.
+   *
+   * Furthermore, there are BIOSes out there that do not implement a HID
+   * descriptor parser, nor do they explicitly send a SET_PROTOCOL request to
+   * put the keyboard into boot protocol. For these to work, we need to default
+   * to boot protocol.
```

Comment:
The usb hid spec actually states that BIOSes typically do not parse report descriptors at all.

<img width="547" alt="image" src="https://user-images.githubusercontent.com/6299/40918126-5ac05e1c-6840-11e8-9383-93fb1c814655.png">

and that usb hid aware OSs should switch to Report manually:
<img width="535" alt="image" src="https://user-images.githubusercontent.com/6299/40918177-81603efc-6840-11e8-9b15-b058555a9892.png">

Where does it say keyboard should default to Report?

Reply:
Section 7.2.6, `Set_Protocol` request:

> When initialized, all devices default to report protocol. However the host should not make any assumptions about the device’s state and should set the desired protocol whenever initializing a device

Reply:
Seems like it's contradicting itself then

Reply:
Couldn't the nkro keyboard prefix the bitmap report with a BOOT compatible 6 key report? Eliminating the need for 2 endpoints

Reply:
Typically BOOT protocol implementations...don't deal well with weirdness. 

(And I read that as the spec saying "clients should do X. hosts should do Y, defensively" -- not exactly contradicting each other so much as being sensible ;)

Reply:
> Couldn't the nkro keyboard prefix the bitmap report with a BOOT compatible 6 key report? Eliminating the need for 2 endpoints

IIRC someone attempted to do that. It probably works for BIOSes and whatnot that ignore junk at the end of the report, but I'm sure there are things out there that doesn't. Two endpoints feel a lot safer to me.